### PR TITLE
IHander getResponse method now takes a HttpRequest

### PR DESCRIPTION
**Pull request by td-extreme**
This is change is made so that custom made handlers can have access to
the HttpRequest that they are handling.

### CODE REVIEW

File: src/main/java/httpConnectionProcessing/HttpConnectionToProcess.java [hunk 1]
```diff
@@ -26,22 +28,28 @@ public void execute() {
...
     }
     catch (InvalidHttpRequestException e) {
       e.printStackTrace();
+      return null;
+    }
+  }
+
+  private IHandler getHandler(HttpRequest request) throws BadConnectionException {
+    IHandler handler;
+    if (request == null) {
```
Comment (mikeknep) [inline src/main/java/httpConnectionProcessing/HttpConnectionToProcess.java:43]: Bummer that you have to add this check, but I can't think of a great alternative.

File: src/test/java/handlers/HandlerPostFileContentsTest.java [hunk 2]
```diff
@@ -9,44 +10,47 @@
...
-    path1 = "testFile-1.txt";
-    path2 = "testFile-2.txt";
-    handler = new HandlerPostFileContents(path, body, mockFileIO);
+    path = "/testFile.txt";
+    path1 = "/testFile-1.txt";
+    path2 = "/testFile-2.txt";
+    handler = new HandlerPostFileContents(mockFileIO);
+    request = new HttpRequest("POST /testFile.txt HTTP/1.1", new HashMap<String, String>());
+    System.out.println("******");
+    System.out.println(request.path());
```
Comment (mikeknep) [inline src/test/java/handlers/HandlerPostFileContentsTest.java]: Pretty sure these two shouldn't be committed.

File: src/main/java/httpConnectionProcessing/HttpConnectionToProcess.java (hunk 1 above)
Comment (mikeknep) [inline src/main/java/httpConnectionProcessing/HttpConnectionToProcess.java:43]: @dadarek any ideas?

File: src/test/java/handlers/HandlerPostFileContentsTest.java (hunk 2 above)
Comment (td-extreme) [inline src/test/java/handlers/HandlerPostFileContentsTest.java]: no, they should not.

File: src/main/java/httpConnectionProcessing/HttpConnectionToProcess.java (hunk 1 above)
Comment (dadarek) [inline src/main/java/httpConnectionProcessing/HttpConnectionToProcess.java:43]: Yeah - why is `request` ever null? If it's null, some class above this already has this knowledge so maybe do something different there?

**Comment by dadarek** [comment]
That null check isn't a big deal, but _maybe_ you can do something about it.

Otherwise +1

File: src/main/java/httpConnectionProcessing/HttpConnectionToProcess.java (hunk 1 above)
Comment (mikeknep) [inline src/main/java/httpConnectionProcessing/HttpConnectionToProcess.java:43]: @dadarek Looks like previously `getRequest` was not a separate method and instead was inlined as part of `getHandler`. The request could be invalid, in which case an exception was raised and caught, leading to returning the "Bad Request" handler. However, now the request is grabbed first and passed to this method. But the original HTTP request could still be invalid, and probably shouldn't be coerced into an `HttpRequest` object if it's gibberish...

Comment (dadarek) [inline src/main/java/httpConnectionProcessing/HttpConnectionToProcess.java:43]: Exactly - how can it still be "invalid" if it gets to this point?

**Pull request merged**